### PR TITLE
build: automate requirements installation and use venv to install requirements only locally

### PR DESCRIPTION
**Pull request by pkalliok-aiven**
This simply takes one step away from the build process.

### CODE REVIEW

**Comment by pkalliok-aiven** [comment]
Ah.  It seems to break the netlify build (for some reason, setting up a virtualenv in netlify is broken).  I can investigate but wouldn't want to put too much time to this :/

File: Makefile [hunk 1]
```diff
@@ -4,50 +4,63 @@
...
 # Put it first so that "make" without argument is like "make help".
-help:
+help: requirements-stamp
 	@$(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
 
-.PHONY: help Makefile
+.PHONY: help Makefile livehtml livehtmlall spell create-index index-devportal index-helpcenter create-feedback-table
+
+myenv:
+	python -m venv myenv
```
Comment (vladan) [inline Makefile:22]: maybe be explicit with `python3` here? (not sure if it's the default in macos)

File: Makefile [hunk 2]
```diff
@@ -4,50 +4,63 @@
...
+
+requirements-stamp: requirements.txt myenv
+	@echo Installing sphinx dependencies...
+	./myenv/bin/pip install -r requirements.txt
```
Comment (vladan) [inline Makefile:26]: not sourcing the env here might not set the python path correctly

**Comment by lornajane** [comment]
I'd be super happy to have this moved to the `README` so that people could choose to use it, or if they wanted to do something different on their setup, or it's being used by netlify then the makefile doesn't make assumptions but the instructions are right there if needed.

**Comment by lornajane** [comment]
I'm abandoning this but additional instructions for the `README` would be welcome (let's leave the build alone since Netlify does this its own way)

**Pull request closed without merging**